.gitlab-ci.yml

```diff
@@ -0,0 +1,179 @@
+default:
+  image: python:3.7
+
+
+#cache:
+#  directories:
+#    - $HOME/venv
+#    - $HOME/.cache/pip
+#    - $HOME/Library/Caches/pip
+#    - $HOME/Library/Caches/Homebrew
+#    - $TRAVIS_BUILD_DIR/.tox
+
+
+stages:
+  - test
+  - build
+  - release
+
+
+test:lint:
+  stage: test
+  script:
+    - make install
+    - make lint
+
+test:lbry-unit:
+  stage: test
+  script:
+    - make install
+    - cd lbry && HOME=/tmp coverage run -p --source=lbry -m unittest discover -vv tests.unit
+
+
+test:lbry-integ:
+  stage: test
+  script:
+    - pip install coverage tox-travis
+    - cd lbry && tox
+
+
+.torba-tests: &torba_tests
+  stage: test
+  script:
+    - pip install coverage tox-travis
+    - cd torba && tox
+
+test:torba-unit:
+  before_script:
+    - export TESTTYPE=unit
+  <<: *torba_tests
+
+test:torba-integ:
+  before_script:
+    - export TESTTYPE=integration
+  <<: *torba_tests
+
+test:json-api:
+  stage: test
+  script:
+    - make install
+    - cd lbry && HOME=/tmp coverage run -p --source=lbry scripts/generate_json_api.py
+
+
+
+.build:
+  stage: build
+  artifacts:
+    expire_in: 1 day
+    paths:
+      - lbrynet-${OS}.zip
+  script:
+    - pip install pyinstaller
+    - pip install -e torba/.
+    - python3.7 docker/set_build.py
+    - pip install -e lbry/.
+    - pyinstaller --onefile --name lbrynet lbry/lbry/extras/cli.py
+    - chmod +x dist/lbrynet
+    - zip --junk-paths ${CI_PROJECT_DIR}/lbrynet-${OS}.zip dist/lbrynet  # gitlab expects artifacts to be in $CI_PROJECT_DIR
+    - openssl dgst -sha256 ${CI_PROJECT_DIR}/lbrynet-${OS}.zip | egrep -o [0-9a-f]+$ # get sha256 of asset. works on mac and ubuntu
+    - dist/lbrynet --version
+
+build:linux:
+  extends: .build
+  image: ubuntu:18.04  # cant use python3.7 image - binary won't run on ubuntu
+  variables:
+    OS: linux
+  before_script:
+    - apt-get update
+    - apt-get install -y --no-install-recommends software-properties-common zip curl build-essential
+    - add-apt-repository -y ppa:deadsnakes/ppa
+    - apt-get install -y --no-install-recommends python3.7-dev python3-setuptools python3-wheel
+    - python3.7 <(curl -q https://bootstrap.pypa.io/get-pip.py) # make sure we get pip with python3.7
+
+build:mac:
+  extends: .build
+  tags: [macos] # makes gitlab use the mac runner
+  variables:
+    OS: mac
+    GIT_DEPTH: 5
+    VENV: /tmp/gitlab-lbry-sdk-venv
+  before_script:
+#    - brew upgrade python || true
+    - python3 --version | grep -q '^Python 3\.7\.' # dont upgrade python on every run. just make sure we're on the right Python
```

Comment:
how does it "make sure"? does the build fail if grep doesn't find the expression?

Reply:
exactly. the build will fail if any command fails. grep will fail if it cannot find what it's looking for